addons/mrp/models/mrp_bom.py

```diff
@@ -249,10 +252,16 @@ def check_cycle(v, visited, recStack, graph):
         lines_done = []
         V |= set([product.product_tmpl_id.id])
 
-        bom_lines = [(bom_line, product, quantity, False) for bom_line in self.bom_line_ids]
+        bom_lines = []
+        products_to_search = self.env['product.product']
         for bom_line in self.bom_line_ids:
-            V |= set([bom_line.product_id.product_tmpl_id.id])
-            graph[product.product_tmpl_id.id].append(bom_line.product_id.product_tmpl_id.id)
+            product_id = bom_line.product_id
+            V |= set([product_id.product_tmpl_id.id])
+            graph[product.product_tmpl_id.id].append(product_id.product_tmpl_id.id)
+            bom_lines.append((bom_line, product, quantity, False))
+            products_to_search |= product_id
+        product_boms = self._get_product2bom(products_to_search, bom_type='phantom', picking_type=picking_type or self.picking_type_id)
```

Comment:
You remove the `company_id` argument here. Then `_get_product2bom` will take the `company_id` in context (if there is) before the `company_id` argument (here == to `self.company_id.id`). I am not sure if the `company_id` from context is use somewhere but it is still a behavior change of the method. Or I miss something ?

Reply:
No no, I indeed forgot the company_id param.